src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -575,5 +575,32 @@ private static IEnumerable<PropertyInfo> GetAllProperties(Type type)
 
             return allProperties;
         }
+
+        private static object GetPropertyValue(PropertyInfo property, object instance, IConfiguration config, BinderOptions options)
+        {
+            string propertyName = GetPropertyName(property);
+            return BindInstance(
+                property.PropertyType,
+                property.GetValue(instance),
+                config.GetSection(propertyName),
+                options);
+        }
+
+        private static string GetPropertyName(MemberInfo property)
+        {
+            if (property == null)
+            {
+                throw new ArgumentNullException(nameof(property));
+            }
+
+            string name = property
```

Comment:
I'm sorry about this, but we have been trying to reduce the usage of `Linq` in `Microsoft.Extensions` as that slows down `ASP.NET` startup time. i.e: https://github.com/dotnet/runtime/commit/57c14a2d86b739fb135da5a0e5c65da38dafe552#diff-84cb3681651b32e35aea35fd61c8e249cab2469a38c84fde029cd607f69183c8

Can we do it iterative instead? 

cc: @stephentoub

Reply:
@safern  reverted commit and applied iterative checks.